presto-function-namespace-managers/src/main/java/com/facebook/presto/functionNamespace/mysql/MySqlFunctionNamespaceManager.java

```diff
@@ -149,7 +151,7 @@ public void createFunction(SqlInvokedFunction function, boolean replace)
 
         jdbi.useTransaction(handle -> {
             FunctionNamespaceDao transactionDao = handle.attach(FunctionNamespaceDao.class);
-            Optional<SqlInvokedFunctionRecord> latestVersion = transactionDao.getLatestRecordForUpdate(function.getFunctionId());
+            Optional<SqlInvokedFunctionRecord> latestVersion = transactionDao.getLatestRecordForUpdate(hash(function.getFunctionId()), function.getFunctionId());
```

Comment:
This is a potential security loophole. I can potentially modify other people's function if I manage to come up with a signature with the same hash right? There at least needs to be some verification that the function you get is the function you actually want to get?

Reply:
Both the hash of a function id and the function id itself are passed in. The DAO query still require both function id hash and function id to match. You won't be able to modify that function even if you come up with a function id with the exact same hash.